cranelift/codegen/src/isa/x64/inst/args.rs

```diff
@@ -544,6 +558,16 @@ impl SyntheticAmode {
     pub(crate) fn finalize(&self, state: &mut EmitState, buffer: &mut MachBuffer<Inst>) -> Amode {
         match self {
             SyntheticAmode::Real(addr) => addr.clone(),
+            SyntheticAmode::IncomingArg {
+                offset,
+                stack_args_size,
+            } => {
+                // NOTE: this could be made relatie to RSP by adding additional offsets from the
+                // frame_layout.
+                let diff = state.frame_layout().stack_args_size - stack_args_size
+                    + state.frame_layout().setup_area_size;
+                Amode::imm_reg(i32::try_from(offset + diff).unwrap(), regs::rbp())
+            }
```

Comment:
If we switch `IncomingArg` to holding offsets from the highest address in the incoming argument area, instead of the lowest address, then we don't need to also store the size of the incoming argument area with it. Finalizing such an amode then amounts to finding the highest address in the stack frame (e.g. `rbp` plus the `setup_area_size` and `stack_args_size` from the frame layout), then subtracting the offset that's in `IncomingArg`.